src/db/routes.js

```diff
@@ -2,11 +2,69 @@
 const pathFromUrl = (url, match = '') => {
     return url.substr(url.indexOf(match) + match.length + 1).split('/');
 };
-
+// endpoints to be trimmed from urls
 const apiUrl = 'data/api';
-const apiRegex = /\/data\/api.*/;
+
+// regex to match different endpoints
+const apiRegex = /\/data\/api.*/,
+    uploadRegex = /\/data\/upload.*/,
+    downloadRegex = /\/data\/download.*/,
+    resetRegex = /\/data\/reset.*/;
 
 export default (workbox, db) => {
+    workbox.routing.registerRoute(
+        resetRegex,
+        async () => {
+            let message = null;
+            let ok = true;
+            try {
+                message = await db.delete(true);
+            } catch (err) {
+                console.error(err);
+                message = err.message;
+                ok = false;
+            }
+            return new Response(JSON.stringify({ ok: ok, query: message }));
+        },
+        'DELETE'
+    );
+
+    workbox.routing.registerRoute(
+        downloadRegex,
+        async () => {
+            let message = null;
+            let ok = true;
+            try {
+                message = await db.download();
+            } catch (err) {
+                console.error(err);
+                message = err.message;
+                ok = false;
+            }
+            return new Response(JSON.stringify({ ok: ok, query: message }));
+        },
+        'GET'
+    );
+
+    workbox.routing.registerRoute(
+        uploadRegex,
+        async ({ event }) => {
+            return event.request.json().then(async payload => {
+                let message = null;
+                let ok = true;
+                try {
+                    await db.upload(payload);
+                } catch (err) {
+                    console.error(err);
+                    message = err.message;
+                    ok = false;
+                }
+                return new Response(JSON.stringify({ ok: ok, query: message }));
```

Comment:
The above 3 functions are essentially duplicates with only a few lines different.  Let's refactor to a function you call with params.